components/ProfilePage/Desktop/PreReveal/PreTwitterLocation.tsx

```diff
@@ -1,40 +1,105 @@
+import { FC } from "react"
 import Media from "../../../../shared/Media"
+import Tooltip from "../../../../shared/Tooltip"
+import { useUserProvider } from "../../../../providers/UserProvider"
+import { TwitterLocationProps } from "../../interface"
 
-const PreTwitterLocation = () => (
-  <div className="flex items-center gap-x-[15px]">
-    <div className="flex items-center gap-x-[5px]">
-      <Media
-        type="image"
-        link="/assets/Profile/black_twitter.svg"
-        blurLink="/assets/Profile/black_twitter.png"
-        containerClasses="w-[23px] h-[19px]"
-      />
-      <p className="font-quicksand font-bold text-[22px] leading-[99.3%]">@mckennarhillier</p>
-    </div>
-    <div className="flex items-center gap-x-[5px]">
-      <Media
-        type="image"
-        link="/assets/Profile/black_location.svg"
-        blurLink="/assets/Profile/black_location.png"
-        containerClasses="w-[26px] h-[26px]"
-      />
-      <p className="font-quicksand font-bold text-[22px] leading-[99.3%]">Denver, CO</p>
+const PreTwitterLocation: FC<TwitterLocationProps> = ({
+  handleEditable,
+  isEditable,
+  handleEditedTwitterHandle,
+  editedTwitterHandle,
+  editedLocation,
+  handleEditedLocation,
```

Comment:
optional: instead of passing these as props, you could create a `<UserProfileProvider />`